modules/aspect_rules_js/1.20.1/MODULE.bazel

```diff
@@ -0,0 +1,167 @@
+        "*": [
+            "no-sandbox",
+        ],
+        # If @kubernetes/client-node is not sandboxed, will fail with
+        # ```
+        # src/azure_auth.ts(97,43): error TS2575: No overload expects 2 arguments, but overloads do exist that expect either 1 or 4 arguments.
+        # src/azure_auth.ts(98,34): error TS2575: No overload expects 2 arguments, but overloads do exist that expect either 1 or 4 arguments.
+        # src/gcp_auth.ts(93,43): error TS2575: No overload expects 2 arguments, but overloads do exist that expect either 1 or 4 arguments.
+        # src/gcp_auth.ts(94,34): error TS2575: No overload expects 2 arguments, but overloads do exist that expect either 1 or 4 arguments.
+        # ```
+        # since a `jsonpath-plus@7.2.0` that is newer then the transitive dep `jsonpath-plus@0.19.0` is found outside of the sandbox that
+        # includes typings that don't match the 0.19.0 "any" usage.
+        "@kubernetes/client-node": [],
+        "@figma/nodegit": [
+            "no-sandbox",
+            # Workaround Engflow not honoring requires-network on build actions
+            "no-remote-exec",
+            "requires-network",
+        ],
+        "esbuild": [
+            "no-sandbox",
+            # Workaround Engflow not honoring requires-network on build actions
+            "no-remote-exec",
+            "requires-network",
+        ],
+        "segfault-handler": [
+            "no-sandbox",
+            # Workaround Engflow not honoring requires-network on build actions
+            "no-remote-exec",
+            "requires-network",
+        ],
+    },
+    npmrc = "//:.npmrc",
+    patch_args = {
+        "*": ["-p1"],
+        "@gregmagolan/test-a": ["-p4"],
+    },
+    patches = {
+        "@gregmagolan/test-a": ["//examples/npm_deps:patches/test-a.patch"],
+        "@gregmagolan/test-a@0.0.1": ["//examples/npm_deps:patches/test-a@0.0.1.patch"],
+        "@gregmagolan/test-b": ["//examples/npm_deps:patches/test-b.patch"],
+        "meaning-of-life@1.0.0": ["//examples/npm_deps:patches/meaning-of-life@1.0.0-after_pnpm.patch"],
+    },
+    pnpm_lock = "//:pnpm-lock.yaml",
+    pnpm_version = "7.25.0",
+    public_hoist_packages = {
+        # Instructs the linker to hoist the ms@2.1.3 npm package to `node_modules/ms` in the `examples/npm_deps` package.
+        # Similar to adding `public-hoist-pattern[]=ms` in .npmrc but with control over which version to hoist and where
+        # to hoist it. This hoisted package can be referenced by the label `//examples/npm_deps:node_modules/ms` same as
+        # other direct dependencies in the `examples/npm_deps/package.json`.
+        "ms@2.1.3": ["examples/npm_deps"],
+    },
+    update_pnpm_lock = True,
+    verify_node_modules_ignored = "//:.bazelignore",
+    verify_patches = "//examples/npm_deps/patches:patches",
+)
+
+use_repo(npm, "npm", "npm__rollup__2.70.2", "npm__webpack-bundle-analyzer__4.5.0__bufferutil_4.0.7")
```

Comment:
it's interesting that there's so much dev-only content in this file. Also feels like a risk that this `use_repo` might accidentally bring an `@npm` for users. Ideally we should have some way to check in the presubmit.yml that some external repos were **not** fetched.

Reply:
My understanding is that because we use the `npm` extension with `dev_dependency = True` it shouldn't bring in this repo. Could you confirm, @Wyverald?

We could also try to strip this stuff out before forming the release archive.

Reply:
Everything with `dev_dependency = True` is a no-op outside the root module, so this should be fine. You shouldn't have to have a release process that strips out part of the module file.

Reply:
I see it's up there on the use_extension